go/engine/pgp_push_private.go

```diff
@@ -0,0 +1,218 @@
+	return &PGPPushPrivate{arg}
+}
+
+func getCurrentUserPGPKeys(m libkb.MetaContext) ([]libkb.PGPFingerprint, error) {
+	uid := m.CurrentUID()
+	if uid.IsNil() {
+		return nil, libkb.NewLoginRequiredError("for push/pull of PGP private keys to KBFS")
+	}
+	upk, _, err := m.G().GetUPAKLoader().LoadV2(libkb.NewLoadUserArgWithMetaContext(m).WithUID(uid))
+	if err != nil {
+		return nil, err
+	}
+	var res []libkb.PGPFingerprint
+	for _, key := range upk.Current.PGPKeys {
+		if key.Base.Revocation != nil {
+			continue
+		}
+		res = append(res, libkb.PGPFingerprint(key.Fingerprint))
+	}
+	if len(res) == 0 {
+		return nil, errors.New("The --all flag only works if you have PGP keys linked to your Keybase account")
+	}
+	return res, nil
+}
+
+func getPrivateFingerprints(m libkb.MetaContext, fps []keybase1.PGPFingerprint) ([]libkb.PGPFingerprint, error) {
+	if len(fps) == 0 {
+		return getCurrentUserPGPKeys(m)
+	}
+
+	var ret []libkb.PGPFingerprint
+	for _, fp := range fps {
+		ret = append(ret, libkb.ImportPGPFingerprint(fp))
+
+	}
+	return ret, nil
+}
+
+func simpleFSClient(m libkb.MetaContext) (*keybase1.SimpleFSClient, error) {
+	xp := m.G().ConnectionManager.LookupByClientType(keybase1.ClientType_KBFS)
+	if xp == nil {
+		return nil, libkb.KBFSNotRunningError{}
+	}
+	return &keybase1.SimpleFSClient{
+		Cli: rpc.NewClient(xp, libkb.NewContextifiedErrorUnwrapper(m.G()), nil),
+	}, nil
+}
+
+func (e *PGPPushPrivate) mkdir(m libkb.MetaContext, fs *keybase1.SimpleFSClient, path string) (err error) {
+	opid, err := fs.SimpleFSMakeOpid(m.Ctx())
+	if err != nil {
+		return err
+	}
+	defer fs.SimpleFSClose(m.Ctx(), opid)
+	err = fs.SimpleFSOpen(m.Ctx(), keybase1.SimpleFSOpenArg{
+		OpID:  opid,
+		Dest:  keybase1.NewPathWithKbfs(path),
+		Flags: keybase1.OpenFlags_DIRECTORY,
+	})
+	return err
+}
+
+func (e *PGPPushPrivate) write(m libkb.MetaContext, fs *keybase1.SimpleFSClient, path string, data string) (err error) {
+	opid, err := fs.SimpleFSMakeOpid(m.Ctx())
+	if err != nil {
+		return err
+	}
+	defer fs.SimpleFSClose(m.Ctx(), opid)
+	err = fs.SimpleFSOpen(m.Ctx(), keybase1.SimpleFSOpenArg{
+		OpID:  opid,
+		Dest:  keybase1.NewPathWithKbfs(path),
+		Flags: keybase1.OpenFlags_WRITE,
+	})
+	if err != nil {
+		return err
+	}
+	err = fs.SimpleFSWrite(m.Ctx(), keybase1.SimpleFSWriteArg{
+		OpID:    opid,
+		Offset:  0,
+		Content: []byte(data),
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (e *PGPPushPrivate) link(m libkb.MetaContext, fs *keybase1.SimpleFSClient, file string, link string) (err error) {
+	err = fs.SimpleFSSymlink(m.Ctx(), keybase1.SimpleFSSymlinkArg{
+		Target: file,
+		Link:   keybase1.NewPathWithKbfs(link),
+	})
+	return err
+}
+
+func (e *PGPPushPrivate) remove(m libkb.MetaContext, fs *keybase1.SimpleFSClient, file string) (err error) {
+	opid, err := fs.SimpleFSMakeOpid(m.Ctx())
+	if err != nil {
+		return err
+	}
+	defer fs.SimpleFSClose(m.Ctx(), opid)
```

Comment:
Despite what the avdl documentation says, I suspect this is returning an error, since for a `Remove` call there is no handle that needs to be closed.  The `Wait` below should be enough to finish the operation.

Reply:
I think I did see an error in my logs, now that you mention it.